core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -80,13 +80,13 @@ object Defaults {
   val LogCleanupPolicy = Delete
   val LogCleanerThreads = 1
   val LogCleanerIoMaxBytesPerSecond = Double.MaxValue
-  val LogCleanerDedupeBufferSize = 500 * 1024 * 1024L
+  val LogCleanerDedupeBufferSize = 128 * 1024 * 1024L
```

Comment:
It's probably ok to lower this. Could you include this change in the update doc as Ismael mentioned?